include/ck_tile/ops/batched_transpose/kernel/batched_transpose_kernel.hpp

```diff
@@ -81,47 +81,46 @@ struct BatchedTransposeKernel
 
         static_assert(kMPerThread == 1 && kNPerThread == 1);
 
-        const auto iDim  = blockIdx.z;
+        const auto iDim = blockIdx.z;
+        // dim[0] = W(M,X), dim[1] = H(N,Y)
         const auto x_m_n = [&]() {
             const auto x_dram_naive = make_naive_tensor_view<address_space_enum::global>(
                 static_cast<const Type*>(kargs.p_input) + iDim * kargs.dim_stride,
-                make_tuple(kargs.height, kargs.width),
-                make_tuple(kargs.width, 1),
-                number<kNPerThread>{}, // TODO thread load value
+                make_tuple(kargs.width, kargs.height),
+                make_tuple(1, kargs.width),
+                number<kMPerThread>{},
```

Comment:
by the naming of `y_m_n`, `n` is the fatest dimension (stride=1), so you should use k**M**PerThread here.